devito/operator/operator.py

```diff
@@ -584,6 +590,9 @@ def _prepare_arguments(self, autotune=None, **kwargs):
             if configuration['mpi']:
                 raise ValueError("Multiple Grids found")
         try:
+            # Take biggest grid, i.e discard grids with subdimensions
+            grids = {g for g in grids if not any(d.is_Sub for d in g.dimensions)}
```

Comment:
this needs to be better abstracted in my opinion

I wonder whether we could have a SubGrid object (returned via `__new__`) to capture subsampled grids. And dramatically simplify our lives here

Reply:
My main though is that this needs to be part of the `Function on subdomain` overall, this is a temporary fix for arg processing

Reply:
>This should be enough to filter away the Subdomain grids?

No this will create issue  for example if there is only one "sub-grid" this would end up empty and leads to missing arguments.

But looks like it's a duplicate from above now might be able to remove that one.